src/Xamarin.Android.Tools.AndroidSdk/AndroidSdkInfo.cs

```diff
@@ -185,5 +190,25 @@ public static void DetectAndSetPreferredJavaSdkPathToLatest (Action<TraceLevel,
 			var sdk = CreateSdk (logger);
 			sdk.SetPreferredJavaSdkPath (latestJdk.HomePath);
 		}
+
+		public static string GetPreferredAndroidToolsPath(string androidSdkPath)
```

Comment:
Hopefully the last comment on this PR, and apologies for not seeing this earlier:

Must this method be `static`?  *Most* other methods on this class are *instance* methods, around the `AndroidSdkInfo.AndroidSdkPath` instance property.

I think a better method more "in keeping" with the current design of the class would be:

```csharp
public IEnumerable<string> GetCommandLineToolsPaths (string preferredCommandLineToolsVersion)
{
	if (!string.IsNullOrEmpty (preferredCommandLineToolsVersion)) {
		var preferredDir = Path.Combine (AndroidSdkPath, "cmdline-tools", preferredCommandLineToolsVersion);
		if (Directory.Exists (preferredDir))
			return new[] { preferredDir }.Concat (GetCommandLineToolsPaths (null).Where (p => p != preferredDir));
	}
	return GetCommandLineToolsPaths ();
}

public IEnumerable<string> GetCommandLineToolsPaths ()
{
	var cmdlineToolsDir = Path.Combine (AndroidSdkPath, "cmdline-tools");
	if (Directory.Exists (cmdlineToolsDir)) {
		var latestDir = Path.Combine (cmdlineToolsDir, "latest");
		if (Directory.Exists (latestDir))
			yield return latestDir;
		foreach (var d in SortedSubdirectoriesByVersion (cmdlineToolsDir)) {
			var version = Path.GetFileName (d);
			if (version == "latest")
				continue;
			yield return d;
		}
	}
	var toolsDir = Path.Combine (AndroidSdkPath, "tools");
	if (Directory.Exists (toolsDir)) {
		yield return toolsDir;
	}
}
```